### PR TITLE
82614: KOTS Supports Preflights for Helm and KOTS in One Release

### PR DESCRIPTION
**Pull request by jonquil2002**
SC: https://app.shortcut.com/replicated/story/82614/document-how-kots-handles-preflight-checks-so-that-you-can-support-helm-and-kots-installs-from-the-same-release

Previews: 

For net new vendors - https://deploy-preview-1290--replicated-docs-upgrade.netlify.app/vendor/preflight-helm-defining
Adds note for existing vendors with KOTS installations that use Helm charts: - https://deploy-preview-1290--replicated-docs-upgrade.netlify.app/vendor/preflight-kots-defining

### CODE REVIEW

File: docs/partials/preflights/_preflights-about.mdx [hunk 1]
```diff
@@ -1,3 +1,3 @@
-You define preflight checks based on your application needs. Replicated recommends that you use preflight checks to help ensure that your customers' environments support your application before installing the application. 
+You define preflight checks based on your application needs. Replicated recommends that you use preflight checks to help ensure that your customers' environments support your application before installing the application. Preflights are added to your application package and run automatically before installing the application.
```
Comment (paigecalvert) [inline docs/partials/preflights/_preflights-about.mdx:1]: I don't think this new sentence applies to Helm installations (users need to manually run preflights for Helm installations, right?) 

I'm also not sure I follow "added to your application package"

**Review by paigecalvert — APPROVED** [review]
Looks good! I added some suggestions

File: docs/vendor/preflight-kots-defining.mdx [hunk 2]
```diff
@@ -6,19 +6,19 @@ import PreflightsDefine from "../partials/preflights/_preflights-define.mdx"
 import PreflightsDefineXref from "../partials/preflights/_preflights-define-xref.mdx"
 import PreflightsCrNote from "../partials/preflights/_preflights-cr-note.mdx"
 import AnalyzersNote from "../partials/preflights/_analyzers-note.mdx"
-
+import HelmPreflights from "../partials/preflights/_helm-preflights.mdx"
 
 # Define Preflight Checks for KOTS
 
  This topic provides a basic understanding and some key considerations about preflight checks for Replicated KOTS installations to help guide you in defining them for your application.
```
[the remark this answers is not in the document]
Comment (jonquil2002) [reply docs/vendor/preflight-kots-defining.mdx:13]: Actually, this topic is also for Helm charts that use KOTS/kURL installers too. The other topic is for Helm installations. Lmk if we should talk this through and make sure we're clear.

File: docs/vendor/preflight-helm-defining.mdx [hunk 3]
```diff
@@ -8,7 +8,7 @@ import AnalyzersNote from "../partials/preflights/_analyzers-note.mdx"
 import RedactorsAbout from "../partials/redactors/_redactors-about.mdx"
 import PreflightSbHelmTemplates from "../partials/preflights/_preflight-sb-helm-templates.mdx"
 import ConfigmapNote from "../partials/support-bundles/_configmap-note.mdx"
-
+import HelmPreflights from "../partials/preflights/_helm-preflights.mdx"
 
 # Define Preflight Checks for Helm Installations
```
[the remark this answers is not in the document]
Comment (jonquil2002) [reply docs/vendor/preflight-helm-defining.mdx:13]: No, this is about Helm installations only and I see how this is made confusing by adding this content in. In retrospect, and after more coffee, I see that nothing really has to be said about KOTS in this Helm installation topic. I'll delete it.

**Review by xavpaice — APPROVED** [review]
LGTM.  Do we need to mention the KOTS release that changes this behavior?

File: docs/vendor/preflight-kots-defining.mdx [hunk 4]
```diff
@@ -7,22 +7,29 @@ import PreflightsDefineXref from "../partials/preflights/_preflights-define-xref
...
+<PreflightsAbout/>
 
-Preflights are added to your application package and run automatically before installing the application.
+Preflights checks for KOTS and kURL run automatically before the application is installed.
 
-<PreflightsSpecLocations/>
+Preflight checks are not included by default, so you must enable them. For KOTS entitlements, you add a Preflight custom resource to your release to define preflight checks for standard Kubernetes manifests and Kubernetes Operators.
+
+:::note
+If you have already created a KOTS release using Helm charts and want to support for Helm installations, Replicated recommends defining preflight checks directly in the Helm charts. This lets you maintain one set of preflight specifications for all installation types.
```
Comment (diamonwiggins) [inline docs/vendor/preflight-kots-defining.mdx:25]:
```suggestion
If you have already created a KOTS release using Helm charts and want to add support for Helm installations, Replicated recommends defining preflight checks directly in the Helm charts. This lets you maintain one set of preflight specifications for all installation types. 
```

**Comment by jonquil2002** [comment]
@xavpaice  Yes, good call on adding the v1.101.0 info. Thanks.

@paigecalvert Ready for review.

File: docs/vendor/preflight-kots-defining.mdx [hunk 5]
```diff
@@ -7,22 +7,31 @@ import PreflightsDefineXref from "../partials/preflights/_preflights-define-xref
...
-<PreflightsSpecLocations/>
+Preflight checks are not included by default, so you must enable them. For KOTS entitlements, you add a Preflight custom resource to your release to define preflight checks for standard Kubernetes manifests and Kubernetes Operators.
+
+:::note
+>Introduced in KOTS v1.101.0
+
+If you have already created a KOTS release using Helm charts and want to add support for Helm installations, Replicated recommends defining preflight checks directly in the Helm charts. This lets you maintain one set of preflight specifications for all installation types. 
+
+Preflights specified in the Helm chart override the Preflight custom resource used by KOTS. During installation, if KOTS cannot find preflights specified in the Helm chart archive, then KOTS searches for `kind: Preflight` in the root of the release. See [Define Preflight Checks for Helm Charts](preflight-helm-defining).
+:::
```
Comment (paigecalvert) [inline docs/vendor/preflight-kots-defining.mdx:30]: ^ helpful!

**Review by paigecalvert — APPROVED** [review]

**Pull request merged**